src/Microsoft.MobileBlazorBindings.Core/NativeComponentRenderer.cs

```diff
@@ -130,5 +135,111 @@ internal NativeComponentAdapter CreateAdapterForChildComponent(IElementHandler p
+                    {
+                        throw new InvalidCastException($"Unable to set property {parameter.Key} on object of type '{component.GetType()}' The error was: Specified cast is not valid. ");
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException($"Object of type '{component.GetType()}' does not have a property matching the name '{parameter.Key}'.");
+                }
+            }
+        }
+
+         /// <summary>
+        /// Converts a string into the specified type. If conversion was successful, parsed property will be of the correct type and method will return true.
+        /// If conversion fails it will return false and parsed property will be null.
+        /// This method supports the 8 data types that are valid navigation parameters in Blazor. Passing a string is also safe but will be returned as is because no conversion is neccessary.
+        /// </summary>
+        /// <param name="type"></param>
+        /// <param name="s"></param>
+        /// <param name="result">The parsed object of the type specified. This will be null if conversion failed.</param>
+        /// <returns>True if s was converted successfully, otherwise false</returns>
+        public static bool TryParse(Type type, string s, out object result)
+        {
+            bool success;
+            
+            if (type == typeof(string))
+            {
+                result = s;
+                success = true;
+            }
+            else if(type == typeof(int))
+            {
+                success = int.TryParse(s, out int parsed);
+                result = parsed;
+            }
+            else if(type == typeof(Guid))
+            {
+                success = Guid.TryParse(s, out Guid parsed);
+                result = parsed;
+            }
+            else if (type == typeof(bool))
+            {
+                success = bool.TryParse(s, out bool parsed);
+                result = parsed;
+            }
+            else if (type == typeof(DateTime))
+            {
+                success = DateTime.TryParse(s, out DateTime parsed);
+                result = parsed;
+            }
+            else if (type == typeof(decimal))
+            {
+                success = decimal.TryParse(s, out decimal parsed);
+                result = parsed;
+            }
+            else if (type == typeof(double))
+            {
+                success = double.TryParse(s, out double parsed);
+                result = parsed;
+            }
+            else if (type == typeof(float))
+            {
+                success = float.TryParse(s, out float parsed);
+                result = parsed;
+            }
+            else if (type == typeof(long))
+            {
+                success = long.TryParse(s, out long parsed);
+                result = parsed;
+            }
+            else
+            {
+                result = 5;
```

Comment:
Maybe `null` is better than `5`? Nothing against `5` per se- it's a wonderful number - but I think `null` is more 😁

Reply:
Wait, what, 5? Don't know how that snuck in. It's a null now.